ui/app/AppLayouts/Wallet/views/walletconnect/WalletConnectSDK.qml

```diff
@@ -176,32 +210,117 @@ WebView {
+                    switch(event.name) {
+                        case "session_request":
+                            root.sessionRequestEvent(event.payload)
+                            break
+                        default:
+                            console.error("WC unknown event type: ", event.type)
+                            break
+                    }
+                }
+            })
+        }
+    }
+
     QtObject {
         id: d
 
         property var sessionProposal: null
         property var sessionType: null
         property bool sdkReady: false
 
+        property ListModel pairingsModel: pairings
+
+        onSdkReadyChanged: {
+            if (sdkReady) {
+                d.getPairings()
+            }
+        }
+
+        function populatePairingsModel(pairList) {
+            pairings.clear();
+            for (let i = 0; i < pairList.length; i++) {
+                pairings.append({
+                    active: pairList[i].active,
+                    topic: pairList[i].topic,
+                    expiry: pairList[i].expiry
+                });
+            }
+        }
+
+
         function isWaitingForSdk() {
             return timer.running
         }
 
-
         function generateSdkCall(methodName, paramsStr, successState, errorState) {
-            return "wcResult = {error: null}; try { wc." + methodName  + "(" + paramsStr + ").then((callRes) => { wcResult = {state: " + successState + ", error: null, result: callRes}; }).catch((error) => { wcResult = {state: " + errorState + ", error: error}; }); } catch (e) { wcResult = {state: " + errorState + ", error: \"Exception: \" + e.message}; }; wcResult"
+            return "wcResult = {}; try { wc." + methodName  + "(" + paramsStr + ").then((callRes) => { wcResult = {state: " + successState + ", error: null, result: callRes}; }).catch((error) => { wcResult = {state: " + errorState + ", error: error}; }); } catch (e) { wcResult = {state: " + errorState + ", error: \"Exception: \" + e.message}; }; wcResult"
+        }
+        function requestSdkAsync(jsCode) {
+            root.runJavaScript(jsCode,
+                function(result) {
+                    timer.restart()
+                }
+            )
         }
-        function requestSdk(jsCode) {
+
+        function requestSdk(methodName, paramsStr, successState, errorState) {
+            const jsCode = "wcResult = {}; try { const callRes = wc." + methodName + "(" + (paramsStr ? (paramsStr) : "") + "); wcResult = {state: " + successState + ", error: null, result: callRes}; } catch (e) { wcResult = {state: " + errorState + ", error: \"Exception: \" + e.message};  }; wcResult"
             root.runJavaScript(jsCode,
                 function(result) {
                     timer.restart()
                 }
             )
         }
 
+        function startListeningForEvents() {
+            const jsCode = "
+                try {
+                    async function processWCEvents() {
```

Comment:
@stefandunca why is this `async`?

Reply:
It's a recommended way of running calls without blocking the WebView runtime thread. For this specific call, it is optional; however, I plan to improve on this with my current work, which will enhance the robustness of the webview-qml communication.